internal/app/machined/pkg/system/services/registry/registry.go

```diff
@@ -123,7 +123,9 @@ func (svc *Service) handler(w http.ResponseWriter, req *http.Request) error {
 	if p.isBlob {
 		s = &singleFileStore{root: svc.root, path: "blob"}
 	} else {
-		s = &singleFileStore{root: svc.root, path: filepath.Join("manifests", ref.Name(), "digest")}
+		refName := handleRegistryWithPort(ref, p)
```

Comment:
I wonder if we could do a simple test (cc @DmitriyMV) here in the `registry_test.go` by providing a fake FS in `testdata` with some structure, but e.g. empty files, and verifying all our conversions for `:` work correctly ?

Reply:
added a test